code/dal.py

```diff
@@ -0,0 +1,117 @@
+    If there are multiple csv files present, it selects the appropriate file that meets the <num_emails> requirement
+    and returns the requested number of rows
+    :param num_emails: The number of rows to be returned
+    :param fetch_all: Ignores num_emails & fetches all the emails in the enron dataset. The number of returned rows may
+    be very large """
+    file_path = '../data/'
+    if fetch_all:
+        file_name = 'all_emails.csv'
+        data = np.loadtxt(file_path + file_name, dtype='str', delimiter=',')
+        return data
+    else:
+        file_name = __get_appr_filename(num_emails, file_path)
+        data = np.loadtxt(file_path + file_name, dtype='str', delimiter=',')
+        if len(data) > num_emails:
+            return data[:num_emails]
+        else:
+            return data
+
+
+def load_from_db(num_emails=100, fetch_all=False):
+    """loads the requested number of emails from the database and dumps the file as csv. For this function to
+    run the mysql database must be set up locally.
+    if fetch_all is false, file is stored as <num_emails>_emails.csv
+    if the fetch_all is true, file is stored as all_emails.csv
+    :param num_emails: The number of emails to be fetched from the db
+    :param fetch_all: Will fetch all the emails from db. Large join query that may take some time"""
+    db_conn = MySQLdb.connect(host='localhost', user='root', passwd='mudit1990', db='EnronAHS')
+    file_path = '../data/'
+    if fetch_all:
+        res = __db_query_all(db_conn)
+        file_name = 'all_emails.csv'
+    else:
+        file_name = str(num_emails) + '_emails.csv'
+        res = __db_query_partial(db_conn, num_emails)
+    data = __clean_data(res)
+    data = data[:,1:] # remove the first column that contains the random id
+    np.savetxt(file_path+file_name, data, fmt='%s', delimiter=',')
+
+
+def __get_appr_filename(num_emails, file_path):
+    """
+    gets the appropriate csv file based on the num_emails required. Tries to select the file with minimum rows that
+    meets the requirement
+    :param num_emails:
+    :param file_path:
+    :return:
+    """
+    files = [f for f in os.listdir(file_path) if os.path.isfile(os.path.join(file_path, f))]
+    max_rows = -1
+    rec_rows = -1
+    for f in files:
+        f_list = f.split('_')
+        try:
+            rows = int(f_list[0])
+            if rows > max_rows:
+                max_rows = rows
+            if rows >= num_emails and (rows < rec_rows or rec_rows == -1):
+                rec_rows = rows
+        except ValueError:
+            continue
+    if rec_rows == -1 and max_rows == -1:
+        return None
+    if rec_rows == -1:
+        rec_rows = max_rows
+    return str(rec_rows) + '_emails.csv'
+
+
+def __clean_data(r):
+    """
+    The function assumes an input as a tuple of tuples, cleans the data and returns a numpy array
+    """
+    data = np.asarray(r)
```

Comment:
Why have we picked just these four characters?
Do you want to look at something along the lines of stopword removal?
Take a look at [this](https://stackoverflow.com/questions/19130512/stopword-removal-with-nltk) and [this](https://pythonspot.com/nltk-stop-words/)

I'm just laying it down here as possible enhancements for the future.

Reply:
yes, this is just preliminary. This layer will support generic cleaning that all model may require including stop words. As of now I just cleaned stuff that would prevent csv read and writes like '\n' and ','